app/controllers/orders/new.js

```diff
@@ -10,10 +10,12 @@ export default Controller.extend({
         for (const attendee of attendees ? attendees.toArray() : []) {
           await attendee.save();
         }
-        if (paymentMode === 'free' || paymentMode === 'bank' || paymentMode === 'cheque' || paymentMode === 'onsite') {
+        if (paymentMode === 'free') {
           order.set('status', 'completed');
-        } else {
+        } else if (paymentMode === 'bank' || paymentMode === 'cheque' || paymentMode === 'onsite') {
```

Comment:
@shreyanshdwivedi What about the online payment orders? They should also come under placed. This code will push them to ending state and error will be generated.

Reply:
@pradeepgangwar it is the demand of the issue to mark online payment orders as `Pending` till the payment is completed. I'm currently fixing the errors which are generated due to this change. Will push them in half an hour.

Reply:
What I see in the code is 

```js
.catch(e => {
            order.set('status', 'pending');
            this.get('notify').error(this.get('l10n').t(` ${e} Oops something went wrong. Please try again`));
          })
          .finally(() => {
            this.set('isLoading', false);
});
```

How will you redirect user to payment page?

Reply:
There is no code block that handles `pending` case.

Reply:
@pradeepgangwar I've updated the files. Now you can see this condition - 
```js
await order.save()
          .then(order => {
            if (order.status === 'pending' || order.status === 'placed') {
              this.get('notify').success(this.get('l10n').t('Order details saved. Please fill the payment details'));
              this.transitionToRoute('orders.placed', order.identifier);
            }
```
There might be a number of bugs still present. I'll have to take a closer look